### PR TITLE
Update Developer Guide with implementation of MarkP command

### LINKED ISSUE #129: Update implementation of MarkP command in Developer Guide

### PR DESCRIPTION
**Pull request by aslam341**
Closes #129 

Also fixed some formatting issues for standardization purposes.

### CODE REVIEW

File: docs/diagrams/MarkPSequenceDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+activate TaskHubParser
+
+create MarkProjectCommand
+TaskHubParser -> MarkProjectCommand
+activate MarkProjectCommand
+
+MarkProjectCommand --> TaskHubParser
+deactivate MarkProjectCommand
+
+TaskHubParser --> LogicManager : c
```
Comment (rayyan35p) [inline docs/diagrams/MarkPSequenceDiagram.puml:27]: Does the `c` stand for command? If so, then maybe future sequence diagrams describing our implementations of commands should follow suit. Existing ones should also be changed to match.

**Review by rayyan35p — CHANGES_REQUESTED** [review]
Good documentation but the changes might need to be moved to the "Proposed Features" section of the DG as I mentioned.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -172,6 +172,28 @@ Step 5. During the execution of the `AddProjectCommand`, selected `Employee` obj
 
 Step 6. A `CommandResult` is produced based on whether the execution was a success or not and returned to the `LogicManager`. 
 
+### Mark Project feature
```
[the remark this answers is not in the document]
Comment (aslam341) [reply docs/DeveloperGuide.md]: Changed it to proposed feature.

File: docs/diagrams/MarkPSequenceDiagram.puml (hunk 1 above)
Comment (aslam341) [reply docs/diagrams/MarkPSequenceDiagram.puml:27]: Yes, `c` stands for command. I have standardized it now.

**Review by rayyan35p — APPROVED** [review]
LGTM!

**Review by Chandan8186 — APPROVED** [review]
LGTM!

**Pull request merged**